lib/ruby_indexer/ruby_indexer.rb

```diff
@@ -0,0 +1,36 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "ruby_indexer/lib/ruby_indexer/visitor"
+require "ruby_indexer/lib/ruby_indexer/index"
+require "ruby_indexer/lib/ruby_indexer/configuration"
+
+module RubyIndexer
+  class << self
+    extend T::Sig
+
+    sig { params(block: T.proc.params(configuration: Configuration).void).void }
+    def configure(&block)
+      block.call(configuration)
+    end
+
+    sig { returns(Configuration) }
+    def configuration
+      @configuration ||= T.let(Configuration.new, T.nilable(Configuration))
+    end
+
+    sig { params(paths: T::Array[String]).returns(Index) }
+    def index(paths = configuration.files_to_index)
```

Comment:
That's a lot of indirection just to pass one value to the indexer.

Why not merging `Configuration` into the `RubyIndexer` and do:

```rb
indexer = RubyIndexer.new(options)
indexer.index(paths)
```

Reply:
Yes, this is done in a follow up PR.